core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -343,8 +364,14 @@ private[spark] class TaskSchedulerImpl(
       if (availableCpus(i) >= CPUS_PER_TASK &&
         resourcesMeetTaskRequirements(availableResources(i))) {
         try {
-          for (task <- taskSet.resourceOffer(execId, host, maxLocality, availableResources(i))) {
+          val (taskOption, didReject) =
+            taskSet.resourceOffer(execId, host, maxLocality, availableResources(i))
+          noDelayScheduleRejects &= !didReject
+          for (task <- taskOption) {
             tasks(i) += task
+            val locality = taskSet.taskInfos(task.taskId).taskLocality
+            minLaunchedLocality = minLaunchedLocality
```

Comment:
shall we just call `minLaunchedLocality = minTaskLocality(minLaunchedLocality, Some(locality))`

Reply:
👍